code/modules/mob/living/carbon/life.dm

```diff
@@ -130,145 +130,199 @@
 		return TRUE
 	return FALSE
 
-
-//Third link in a breath chain, calls handle_breath_temperature()
+/**
+ * This proc tests if the lungs can breathe, if the mob can breathe a given gas mixture, and throws/clears gas alerts.
+ * If there are moles of gas in the given gas mixture, side-effects may be applied/removed on the mob.
+ * This proc expects a lungs organ in order to breathe successfully, but does not defer any work to it.
+ *
+ * Returns TRUE if the breath was successful, or FALSE if otherwise.
+ *
+ * Arguments:
+ * * breath: A gas mixture to test, or null.
+ */
 /mob/living/carbon/proc/check_breath(datum/gas_mixture/breath)
+	. = TRUE
+
 	if(status_flags & GODMODE)
 		failed_last_breath = FALSE
 		clear_alert(ALERT_NOT_ENOUGH_OXYGEN)
-		return FALSE
+		return
+
 	if(HAS_TRAIT(src, TRAIT_NOBREATH))
-		return FALSE
+		return
 
-	var/obj/item/organ/internal/lungs = getorganslot(ORGAN_SLOT_LUNGS)
-	if(!lungs)
-		adjustOxyLoss(2)
+	// Breath may be null, so use a fallback "empty breath" for convenience.
+	if(!breath)
+		breath = empty_breath
```

Comment:
how would we get null air?

Reply:
Space! As you showed me in Discord, `space/remove_air(amount)` returns `null`